command/agent/command.go

```diff
@@ -404,8 +398,21 @@ func (config *Config) verifyUniqueListeners() error {
 func (c *Config) discoverEc2Hosts() ([]string, error) {
 	config := c.EC2Discovery
 	awsConfig := &aws.Config{
-		Region:      aws.String(config.Region),
-		Credentials: credentials.NewStaticCredentials(config.AccessKeyID, config.SecretAccessKey, ""),
+		Region: aws.String(config.Region),
+		Credentials: credentials.NewChainCredentials(
+			[]credentials.Provider{
```

Comment:
Can we please get support for [ECS Task Roles](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html) here? This would allow Consul to fetch credentials meant specifically for it rather than using the instance role when running as a Docker container.

Sample code: https://github.com/aws/aws-sdk-go/blob/master/aws/defaults/defaults.go#L102